doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,430 @@
+
+The difference here is that the decoding of binary and string data does not involve copying of the data to a specified buffer.
+Instead, as described in :ref:`migration_tinycbor_to_zcbor_general`, the decoding function is paired with a ``struct zcbor_string`` type variable where decoding will store position and length of the data, within the provided CBOR buffer.
+It is then up to the user to copy the values to a specified location.
+This mechanism improves the RAM and CPU usage because the need for an intermediate buffer is often reduced.
+
+To process a CBOR buffer and decode it according to the defined mapping, ``zcbor_map_decode_bulk`` takes the following parameters:
+
+ * The zcbor decoding context, ``zsd`` in the above example.
+ * The mapping specification and size of the mapping.
+ * A pointer to a variable for number of successfully decoded map elements.
+
+.. note::
+   Currently, there is no method provided to find out what fields did not get found in string.
+   It is up to the user to verify whether keys that are optional have appeared in the decoded string.
+   This can be done, for example, by checking if a destination variable value has changed from initial.
+
+New Zephyr logging service - logging v2
+***************************************
+
+Zephyr v3.x.x comes with a new logging service (logging v2) that replaces and extends the older one.
+For details about the Zephyr logging service, see :ref:`zephyr:logging_api`.
+Version 2 supports the same set of features with a number of extensions, however, the logging backend API is different.
+All backends in the tree support version 2 API but if you are using an out-of-tree backend, it must be adapted to use the new logging system.
+
+*Required action*: Logging v1 uses the following three functions that need to be replaced:
+
+.. code-block::
+
+   /* DEPRECATED! Functions used for logging v1. */
+   void (*put)(const struct log_backend *const backend,
+            struct log_msg *msg);
+   void (*put_sync_string)(const struct log_backend *const backend,
+            struct log_msg_ids src_level, uint32_t timestamp,
+            const char *fmt, va_list ap);
+   void (*put_sync_hexdump)(const struct log_backend *const backend,
+            struct log_msg_ids src_level, uint32_t timestamp,
+            const char *metadata, const uint8_t *data, uint32_t len);
+
+Replace these functions with the following one function used by logging v2:
+
+.. code-block::
+
+   /* Logging v2 function. */
+   void (*process)(const struct log_backend *const backend,
+           union log_msg2_generic *msg);
+
+Zephyr v3.x.x namespace change
+******************************
+
+All Zephyr public headers have been moved to :file:`include/zephyr`, meaning they need to be prefixed with ``<zephyr/...>`` when included.
+Because this change can potentially break many applications or libraries, ``CONFIG_LEGACY_INCLUDE_PATH`` is provided to allow using the old include path.
```

Comment:
```suggestion
Because this change can potentially break many applications or libraries, :kconfig:option:`CONFIG_LEGACY_INCLUDE_PATH` is provided to allow using the old include path.
```